llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -92,6 +96,36 @@ static cl::opt<bool> SplitEsimd{"split-esimd",
                                 cl::desc("Split SYCL and ESIMD kernels"),
                                 cl::cat(PostLinkCat)};
 
+static cl::opt<bool> LowerEsimd{
+    "lower-esimd", cl::desc("Lower ESIMD constructs"), cl::cat(PostLinkCat)};
+
+static cl::opt<bool>
+    OptLevelO0("O0", cl::desc("Optimization level 0. Similar to clang -O0"),
+               cl::cat(PostLinkCat));
+
+static cl::opt<bool>
+    OptLevelO1("O1", cl::desc("Optimization level 1. Similar to clang -O1"),
+               cl::cat(PostLinkCat));
+
+static cl::opt<bool>
+    OptLevelO2("O2", cl::desc("Optimization level 2. Similar to clang -O2"),
+               cl::cat(PostLinkCat));
+
+static cl::opt<bool> OptLevelOs(
+    "Os",
+    cl::desc(
+        "Like -O2 with extra optimizations for size. Similar to clang -Os"),
+    cl::cat(PostLinkCat));
+
+static cl::opt<bool> OptLevelOz(
+    "Oz",
+    cl::desc("Like -Os but reduces code size further. Similar to clang -Oz"),
+    cl::cat(PostLinkCat));
+
+static cl::opt<bool>
+    OptLevelO3("O3", cl::desc("Optimization level 3. Similar to clang -O3"),
+               cl::cat(PostLinkCat));
```

Comment:
How these are expected to be used? Will they be propagated to the tool based on the same options passed to clang driver?

Reply:
Yes, I will submit the patch for that later and put the link here.

Reply:
Here is the driver portion that will pass the optimization level to sycl-post-link: https://github.com/intel/llvm/pull/3207.